source/SH3/system/msg/msg.cpp

```diff
@@ -0,0 +1,40 @@
+/** @file
+ *  Implementation of msg.hpp
+ *
+ *
+ *  @copyright 2016-2017  Palm Studios
+ *
+ *  @date 16-1-2017
+ *
+ *  @author Jesse Buhagiar
+ */
+#include "SH3/SH3/system/msg/msg.hpp"
+
+using namespace sh3_system;
+
+void sh3_msg::Load(sh3_arc_vfile& file)
+{
+
+    std::uint8_t c;
+    std::uint8_t c2;
+    sh3_arc_vfile::read_error e;
+
```

Comment:
Ah, makes sense. We could reserve some amount representative of one of the longer, average message sizes to avoid some memory allocation overhead. Or since we buffer the whole file anyway we could trivially expose a function that looks for some pattern and returns the offset, but since this might change I'd perhaps just leave it as is and plunk down a `//TODO: can we reserve() some space?`.